swarm/api/http/server_test.go

```diff
@@ -404,52 +399,17 @@ func TestBzzResource(t *testing.T) {
 		t.Fatalf("Expected body '%x', got '%x'", update2Data, b)
 	}
 
-	// get latest update (1.2) with specified period
-	log.Info("get update latest = 1.2")
-	testBzzResUrl = fmt.Sprintf("%s/bzz-resource:/%s/?period=1", srv.URL, correctManifestAddrHex)
-	resp, err = http.Get(testBzzResUrl)
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		t.Fatalf("err %s", resp.Status)
-	}
-	b, err = ioutil.ReadAll(resp.Body)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if !bytes.Equal(update2Data, b) {
-		t.Fatalf("Expected body '%x', got '%x'", update2Data, b)
-	}
+	// test manifest-less queries
+	log.Info("get first update in update1Timestamp via direct query")
+	lp := mru.NewHistoryLookupParams(&updateRequest.View, update1Timestamp, lookup.NoClue)
 
-	// get first update (1.1) with specified period and version
-	log.Info("get first update 1.1")
-	testBzzResUrl = fmt.Sprintf("%s/bzz-resource:/%s?period=1&version=1", srv.URL, correctManifestAddrHex)
-	resp, err = http.Get(testBzzResUrl)
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		t.Fatalf("err %s", resp.Status)
-	}
-	b, err = ioutil.ReadAll(resp.Body)
+	urlq, err := url.Parse(fmt.Sprintf("%s/bzz-resource:/", srv.URL))
 	if err != nil {
 		t.Fatal(err)
 	}
-	if !bytes.Equal(update1Data, b) {
-		t.Fatalf("Expected body '%x', got '%x'", update1Data, b)
-	}
 
-	// test manifest-less queries
-	log.Info("get first update 1.1 via direct query")
-	urlq, err := url.Parse(fmt.Sprintf("%s/bzz-resource:/?period=1&version=1", srv.URL))
-	if err != nil {
-		t.Fatal(err)
-	}
 	values := urlq.Query()
-	updateRequest.View.ToValues(values) // this adds view query parameters
+	lp.ToValues(values) // this adds view query parameters
```

Comment:
This is a property bag serialization pattern. The query string could contain other parameters and we just want to add/read the ones interesting to us.
I have renamed the method to `AppendValues` to make it more explicit.

Reply:
I'm not sure I understand fully.

My point merely was that I was to expecting to see something like `lp.Encode()` instead of `values.Encode()` below. How does this approach make the code clearer?

Reply:
Actually, no. If we have our package encode the values directly, we prevent other possible code to add its own query string parameters.

Reply:
Yes, I see. .. or something like `lp.GetValues()`?